src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9055,9 +9055,23 @@ private ExpressionSyntax ParseSubExpressionCore(Precedence precedence)
             {
                 var questionToken = this.EatToken();
                 var colonLeft = this.ParsePossibleRefExpression();
-                var colon = this.EatToken(SyntaxKind.ColonToken);
-                var colonRight = this.ParsePossibleRefExpression();
-                leftOperand = _syntaxFactory.ConditionalExpression(leftOperand, questionToken, colonLeft, colon, colonRight);
+                if (this.CurrentToken.Kind == SyntaxKind.EndOfFileToken && this.lexer.FollowedByColon)
+                {
+                    // We have an interpolated string with an interpolation that contains a conditional expression.
+                    // Unfortunately, the precedence demands that the colon is considered part of the interpolated
```

Comment:
this always seems weird for me.  Why was precedence chosen this way.  It seems like we'd want the : for interpolations to have the lowest precedence.  Allowing for a nested conditional without issue, and also allowing for something like:

```$"Foo is ${x ? y : z :00}"```

Is there a benefit to the precedence that was chosen?  (And if there is no benefit, would it be a breaking change to change it?  Or would is just allow more programs we currently disallow).

Thanks!

Reply:
The short answer is that the separation of an interpolated string into its parts is done during lexical analysis, long before precedence has any meaning in parsing. That is an implementation-oriented explanation, but there is an isomorphic explanation in terms of the structure of the language specification.

Reply:
Ah, that makes sense.  I can see now how that would affect things.  Oh well, it's doubtful that it would be that beneficial to change this.  Plus, it would likely massively complicate the current relationship between the parser and lexer.  Thanks!